packages/crawler/src/storage/local-data-store.ts

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import Apify from 'apify';
+
+import { DataStore } from './store-types';
+
+export class LocalDataStore implements DataStore {
+    constructor(public readonly storeName: string, private datasetStore?: Apify.Dataset, private readonly apifyObj: typeof Apify = Apify) {}
+
+    public async pushData(data: object | object[]): Promise<void> {
+        await this.open().then(async () => {
```

Comment:
The 
```
await this.open();
await this.datasetStore.pushData(data);
```
should be sufficient without then() chain

Reply:
then does not stops the execution flow and runs next function in the chain only when the previous promise is finished, maybe it's not needed as you mentioned but I usually prefer it, we can revert it tho